### ISSUE TITLE
Does Nested/Transparent Gateway support child device (Azure IoT SDK) authenticate with Azure IoT Hub by CA signed X.509?

### ISSUE THREAD

**Issue by TPE-TIGER** [Azure/iotedge#6809]
Hi, 
I am testing Transparent Gateway's capability, and found Transparent Gateway doens't support CA signed X.509 which come from child device, like below diagram. Can you confirm it?

![image](https://user-images.githubusercontent.com/98569390/206103206-c03b836d-b10d-49b1-9575-357af7efd12a.png)

## Expected Behavior
Not sure does this authenticatoin mode support or not?

## Current Behavior
Tell us what happens instead of the expected behavior

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. 
2.
3.
4.

### Output of `iotedge check`

<details>
<summary>Click here</summary>

```

<Paste here between the triple backticks>

```
</details>

### Device Information
* Host OS [e.g. Ubuntu 18.04, Windows Server IoT 2019]:  Ubuntu 20.04
* Architecture [e.g. amd64, arm32, arm64]: amd64
* Container OS [e.g. Linux containers, Windows containers]: Linux

### Runtime Versions
* aziot-edged [run `iotedge version`]: iotedge 1.4.3
* Edge Agent [image tag (e.g. 1.0.0)]: 1.4
* Edge Hub [image tag (e.g. 1.0.0)]: 1.4
* Docker/Moby [run `docker version`]: 20.10.21+azure-1

Note: when using Windows containers on Windows, run `docker -H npipe:////./pipe/iotedge_moby_engine version` instead

## Logs

<details>
<summary>aziot-edged logs</summary>

```

<Paste here between the triple backticks>

```
</details>

<details>
<summary>edge-agent logs</summary>

```

<Paste here between the triple backticks>

```
</details>

<details>
<summary>edge-hub logs</summary>

```

<4> 2022-12-07 05:46:07.180 +00:00 [WRN] - Error validating cert with Subject: System.Security.Cryptography.X509Certificates.X500DistinguishedName Thumbprint: 099D4647CD5267AFE3E8DF6E6E95A5F1207B86EF52139DFDECADFC4168D5CAD4
<4> 2022-12-07 05:46:07.181 +00:00 [WRN] - Invalid certificate with subject CN=moxa-ivr-01, O=Internet Widgits Pty Ltd, L=Taipei, S=Taiwan, C=TW for moxa-ivr-01
<6> 2022-12-07 05:46:07.183 +00:00 [INF] - Unable to authenticate client moxa-ivr-01 with cached service identity moxa-ivr-01 (Found: True). Resyncing service identity...
<6> 2022-12-07 05:46:07.188 +00:00 [INF] - Refreshing service identity for moxa-ivr-01
<6> 2022-12-07 05:46:07.372 +00:00 [INF] - Not changed node: moxa-ivr-01
<4> 2022-12-07 05:46:07.376 +00:00 [WRN] - Error validating cert with Subject: System.Security.Cryptography.X509Certificates.X500DistinguishedName Thumbprint: 099D4647CD5267AFE3E8DF6E6E95A5F1207B86EF52139DFDECADFC4168D5CAD4
<4> 2022-12-07 05:46:07.376 +00:00 [WRN] - Invalid certificate with subject CN=moxa-ivr-01, O=Internet Widgits Pty Ltd, L=Taipei, S=Taiwan, C=TW for moxa-ivr-01
<6> 2022-12-07 05:46:07.379 +00:00 [INF] - Client moxa-ivr-01 in device scope not authenticated locally.
<3> 2022-12-07 05:46:07.383 +00:00 [ERR] - Unable to generate identity for clientId moxa-ivr-01 and username TPE-TIGER.azure-devices.net/moxa-ivr-01/?api-version=2019-10-01&DeviceClientType=iothubclient%2f1.4.1%20(native%3b%20Linux%3b%20armv7l)
<6> 2022-12-07 05:46:07.384 +00:00 [INF] - "ClientNotAuthenticated, Client ID: moxa-ivr-01; Username: TPE-TIGER.azure-devices.net/moxa-ivr-01/?api-version=2019-10-01&DeviceClientType=iothubclient%2f1.4.1%20(native%3b%20Linux%3b%20armv7l), 6916490c"
<4> 2022-12-07 05:46:08.013 +00:00 [WRN] - Error validating cert with Subject: System.Security.Cryptography.X509Certificates.X500DistinguishedName Thumbprint: 099D4647CD5267AFE3E8DF6E6E95A5F1207B86EF52139DFDECADFC4168D5CAD4
<4> 2022-12-07 05:46:08.013 +00:00 [WRN] - Invalid certificate with subject CN=moxa-ivr-01, O=Internet Widgits Pty Ltd, L=Taipei, S=Taiwan, C=TW for moxa-ivr-01
<6> 2022-12-07 05:46:08.013 +00:00 [INF] - Unable to authenticate client moxa-ivr-01 with cached service identity moxa-ivr-01 (Found: True). Resyncing service identity...
<6> 2022-12-07 05:46:08.014 +00:00 [INF] - Skipping refresh for identity: moxa-ivr-01, waiting 119 seconds until refreshing again.
<4> 2022-12-07 05:46:08.016 +00:00 [WRN] - Error validating cert with Subject: System.Security.Cryptography.X509Certificates.X500DistinguishedName Thumbprint: 099D4647CD5267AFE3E8DF6E6E95A5F1207B86EF52139DFDECADFC4168D5CAD4
<4> 2022-12-07 05:46:08.016 +00:00 [WRN] - Invalid certificate with subject CN=moxa-ivr-01, O=Internet Widgits Pty Ltd, L=Taipei, S=Taiwan, C=TW for moxa-ivr-01
<6> 2022-12-07 05:46:08.017 +00:00 [INF] - Client moxa-ivr-01 in device scope not authenticated locally.
<3> 2022-12-07 05:46:08.019 +00:00 [ERR] - Unable to generate identity for clientId moxa-ivr-01 and username TPE-TIGER.azure-devices.net/moxa-ivr-01/?api-version=2019-10-01&DeviceClientType=iothubclient%2f1.4.1%20(native%3b%20Linux%3b%20armv7l)
<6> 2022-12-07 05:46:08.019 +00:00 [INF] - "ClientNotAuthenticated, Client ID: moxa-ivr-01; Username: TPE-TIGER.azure-devices.net/moxa-ivr-01/?api-version=2019-10-01&DeviceClientType=iothubclient%2f1.4.1%20(native%3b%20Linux%3b%20armv7l), 5c6e0466"
<4> 2022-12-07 05:49:17.220 +00:00 [WRN] - Error validating cert with Subject: System.Security.Cryptography.X509Certificates.X500DistinguishedName Thumbprint: 8102891E7A396B62B74DC43C8EE197E131C4CDD3E22E6F4F7DEF6541660EDD35
<4> 2022-12-07 05:49:17.221 +00:00 [WRN] - Invalid certificate with subject CN=moxa-ivr-01 for moxa-ivr-01
<6> 2022-12-07 05:49:17.221 +00:00 [INF] - Unable to authenticate client moxa-ivr-01 with cached service identity moxa-ivr-01 (Found: True). Resyncing service identity...
<6> 2022-12-07 05:49:17.221 +00:00 [INF] - Refreshing service identity for moxa-ivr-01


```
</details>

## Additional Information
Please provide any additional information that may be helpful in understanding the issue.

**Comment by Maci3jPy** [comment]
Hi, 
I am having the working setup with downstream device authentication as CA signed and transparent gateway. But indeed, it works weird and not how the documentation states. Because it looks like the sensor provision directly to Edge. So, you don't even need to have the certificate in the Hub. But device certificate needs to be created from the same root as the edge certificate and the edge certificate needs to be full chain. Then it looks like it authenticates against the certificates on Edge.

**Comment by TPE-TIGER** [comment]
Hi Maci3jPy,
Thanks your interesting approach. With your way, does child device (sensor) still belong to "device" of Azure IoT Hub? The answer shall be No, right?

**Comment by Maci3jPy** [comment]
@TPE-TIGER 
The device needs to be a child of the specified gateway.

**Comment by brotherneeru** [comment]
@TPE-TIGER IoTedge devices as transparent gateways utilizing CA signed X509 is supported for downstream devices. Please refer to the docs [here] (https://learn.microsoft.com/en-us/azure/iot-edge/iot-edge-as-gateway?view=iotedge-1.4) as well as follow the CA-signed x509 track [here](https://learn.microsoft.com/en-us/azure/iot-edge/how-to-create-transparent-gateway?view=iotedge-1.4&tabs=iotedge)

**Comment by TPE-TIGER** [comment]
Hi Brotherneeru,
Thanks your comment, but you misunderstand my problem.
I have no doubt to use CA signed certificate on downstream certificate. The problem is child device credential which present to Azure IoT Hub. 
When you create a new "device" on Azure IoT Hub, the credentail could be 1) Symmetirc Key, 2) self-signed X.509, and 3) CA signed X.509. (P.S. the credential here has nothing with downstream certificate). My question is when there is one transparent gateway located between the "device" and Azure IoT Hub, does transparent gateway able to handle 3) CA signed X.509?

**Comment by brotherneeru** [comment]
@TPE-TIGER I dont want to confuse you with my terminology. A child device referred to in our transparent gateway can be a downstream device or a gateway device in a nested topology. In your case, it is synonymous with the downstream device referred to in the docs. 

If you follow through that documentation, you will reach this [page](https://learn.microsoft.com/en-us/azure/iot-edge/how-to-authenticate-downstream-device?view=iotedge-1.4). In that page, you will find 3 sections, one for child devices/downstream devices authenticating with CA signed x509. 

In short, yes transparent gateways can support CA signed X.509.

**Comment by TPE-TIGER** [comment]
Hi brotherneeru,
Thanks your answer. It is celar to me now.
